src/components/calcite-dropdown/calcite-dropdown.tsx

```diff
@@ -0,0 +1,237 @@
+import {
+  Component,
+  Element,
+  h,
+  Host,
+  Listen,
+  Prop,
+  State
+} from "@stencil/core";
+import {
+  UP,
+  DOWN,
+  TAB,
+  ENTER,
+  ESCAPE,
+  HOME,
+  END,
+  SPACE
+} from "../../utils/keys";
+import { getElementDir } from "../../utils/dom";
+import { guid } from "../../utils/guid";
+
+@Component({
+  tag: "calcite-dropdown",
+  styleUrl: "calcite-dropdown.scss",
+  shadow: true
+})
+export class CalciteDropdown {
+  //--------------------------------------------------------------------------
+  //
+  //  Element
+  //
+  //--------------------------------------------------------------------------
+
+  @Element() el: HTMLElement;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Public Properties
+  //
+  //--------------------------------------------------------------------------
+
+  @State() active: boolean = false;
+
+  /** specify the alignment of dropdrown, defaults to left */
+  @Prop({ mutable: true, reflect: true }) alignment:
+    | "left"
+    | "right"
+    | "center" = "left";
+
+  /** specify the alignment of dropdrown, defaults to left */
+  @Prop({ mutable: true, reflect: true }) theme: "light" | "dark" = "light";
+
+  /** specify the scale of dropdrown, defaults to m */
+  @Prop({ mutable: true, reflect: true }) scale: "s" | "m" | "l" = "m";
+
+  //--------------------------------------------------------------------------
+  //
+  //  Lifecycle
+  //
+  //--------------------------------------------------------------------------
+
+  connectedCallback() {
+    // validate props
+    let alignment = ["left", "right", "center"];
+    if (!alignment.includes(this.alignment)) this.alignment = "left";
+
+    let theme = ["light", "dark"];
+    if (!theme.includes(this.theme)) this.theme = "light";
+
+    let scale = ["s", "m", "l"];
```

Comment:
Should we extract this out to a util function?

Reply:
I would like to at some point, I use these "prop validations" all over the place. Should discuss further in this issue from awhile back - https://github.com/Esri/calcite-components/issues/45 - can handle outside of PR, though I do agree we should.